kubeconfig/kubeconfig.go

```
@@ -0,0 +1,175 @@
+	Clusters       []Clusters `yaml:"clusters"`
+	Users          []Users    `yaml:"users"`
+	Contexts       []Contexts `yaml:"contexts"`
+	CurrentContext string     `yaml:"current-context"`
+	Kind           string     `yaml:"kind"`
+}
+
+type Clusters struct {
+	Cluster ClusterInfo `yaml:"cluster"`
+	Name    string      `yaml:"name"`
+}
+
+type ClusterInfo struct {
+	CertificateAuthority string `yaml:"certificate-authority"`
+	Server               string `yaml:"server"`
+}
+
+type Users struct {
+	Name string `yaml:"name"`
+	User User   `yaml:"user"`
+}
+
+type User struct {
+	Token string `yaml:"token"`
+}
+
+type Contexts struct {
+	Context Context `yaml:"context"`
+	Name    string  `yaml:"name"`
+}
+
+type Context struct {
+	Cluster   string `yaml:"cluster"`
+	Namespace string `yaml:"namespace"`
+	User      string `yaml:"user"`
+}
+
+func generateKubeConfig(token, server, namespace string) *KubeConfig {
+	return &KubeConfig{
+		APIVersion: "v1",
+		Clusters: []Clusters{
+			{
+				Cluster: ClusterInfo{
+					CertificateAuthority: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+					Server:               server,
+				},
+				Name: server,
+			},
+		},
+		Users: []Users{
+			{
+				Name: "developer",
+				User: User{
+					Token: token,
+				},
+			},
+		},
+		Contexts: []Contexts{
+			{
+				Context: Context{
+					Cluster:   server,
+					Namespace: namespace,
+					User:      "developer",
+				},
+				Name: "developer-context",
+			},
+		},
+		CurrentContext: "developer-context",
+		Kind:           "Config",
+	}
+}
+
+func createKubeConfigText(token, namespace string) string {
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	if host == "" || port == "" {
+		return ""
+	}
+
+	server := "https://" + net.JoinHostPort(host, port)
+	kubeconfig := generateKubeConfig(token, server, namespace)
+
+	bytes, err := yaml.Marshal(&kubeconfig)
+	if err != nil {
+		logrus.Error("error: %v", err)
+		return ""
```

Wouldn't it better to propagate error here instead of initializing empty kubeconfig?